core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java

```diff
@@ -150,120 +148,134 @@ public void testOutputForSimplePage()
     @Test
     public void testOutputForEmptyPage()
     {
-        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT).build();
+        PartitionedOutputOperator partitionedOutputOperator = pagePartitioner(BIGINT).buildPartitionedOutputOperator();
         Page page = new Page(createLongsBlock(ImmutableList.of()));
 
-        processPages(partitionedOutputOperator, page);
+        partitionedOutputOperator.addInput(page);
+        partitionedOutputOperator.finish();
 
         List<Object> partitioned = readLongs(outputBuffer.getEnqueuedDeserialized(), 0);
         assertThat(partitioned).isEmpty();
     }
 
-    @Test
-    public void testOutputForPageWithNoBlockPartitionFunction()
+    @Test(dataProvider = "partitioningMode")
+    public void testOutputForSimplePage(PartitioningMode partitioningMode)
+    {
+        PagePartitioner pagePartitioner = pagePartitioner(BIGINT).build();
+        Page page = new Page(createLongSequenceBlock(0, POSITIONS_PER_PAGE));
+        List<Object> expected = readLongs(Stream.of(page), 0);
+
+        processPages(pagePartitioner, partitioningMode, page);
+
+        List<Object> partitioned = readLongs(outputBuffer.getEnqueuedDeserialized(), 0);
+        assertThat(partitioned).containsExactlyInAnyOrderElementsOf(expected); // order is different due to 2 partitions joined
```

Comment:
> // order is different due to 2 partitions joined

what does that mean? You called `getEnqueuedDeserialized(0)`, so you get data for partition 0 only

Reply:
`readLongs(outputBuffer.getEnqueuedDeserialized(), 0); ` mean extract the data from channel 0 for all partitions